src/media/js/addon/actions/addon.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+import {createAction} from 'redux-actions';
+import req from 'request';
+import Url from 'urlgray';
+import urlJoin from 'url-join';
+
+
+export const FETCH_OK = 'ADDON_ADDON__FETCH_OK';
+const fetchOk = createAction(FETCH_OK);
+
+export const FETCH_VERSIONS_OK = 'ADDON_ADDON__FETCH_VERSIONS_OK';
+const fetchVersionsOk = createAction(FETCH_VERSIONS_OK);
+
+
+export function fetch(addonSlug) {
+  /*
+    Fetch add-on.
+  */
+  return (dispatch, getState) => {
+    const apiArgs = getState().apiArgs || {};
+    const addonUrl = Url(
+      urlJoin(process.env.MKT_API_ROOT, 'extensions/extension/', addonSlug)
+    ).q(apiArgs);
+
+    if (process.env.NODE_ENV === 'test') {
```

Comment:
I think I misunderstood where you wanted to put that originally.

That seems pretty good for some basic GET cases but I think it would get hairy when you start adding failure cases and POST/PUT/DELETE requests.